### PR TITLE
Simplify ignore_malformed handling for synthetic souce in aggregate_metric_double

### PR DESCRIPTION
**Pull request by lkts**
This PR reworks the logic to follow the pattern from #109882. This also fixes the edge case of array of values some of which are malformed.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/CompositeSyntheticFieldLoader.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+import static java.util.Collections.emptyList;
+
+/**
+ * A {@link SourceLoader.SyntheticFieldLoader} that uses a set of sub-loaders
+ * to produce synthetic source for the field.
+ * Typical use case is to gather field values from doc_values and append malformed values
+ * stored in a different field in case of ignore_malformed being enabled.
+ */
+public class CompositeSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
```
Comment (lkts) [inline server/src/main/java/org/elasticsearch/index/mapper/CompositeSyntheticFieldLoader.java:30]: This is all from #109882.

**Comment by lkts** [comment]
@elasticmachine update branch

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java [hunk 2]
```diff
@@ -677,24 +672,20 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
...
+                        // There is a malformed value, but it is not an object (since subParser is null).
+                        // So we just need to copy this single value.
+                        malformedDataForSyntheticSource = XContentBuilder.builder(context.parser().contentType().xContent())
+                            .copyCurrentStructure(context.parser());
+                    }
+                }
+
+                if (malformedDataForSyntheticSource != null) {
+                    context.doc().add(IgnoreMalformedStoredValues.storedField(name(), malformedDataForSyntheticSource));
                 }
```
Comment (kkrik-es) [inline x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:689]: This blocks gets repeated, consider moving it to a helper in `IgnoreMalformedStoredValues`

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java [hunk 3]
```diff
@@ -724,11 +715,15 @@ protected SyntheticSourceMode syntheticSourceMode() {
...
     @Override
     public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
-        // Note that malformed values are handled via `IgnoredSourceFieldMapper` infrastructure
-        return new AggregateMetricSyntheticFieldLoader(name(), simpleName(), metrics);
+        return new CompositeSyntheticFieldLoader(
+            simpleName(),
+            name(),
+            new AggregateMetricSyntheticFieldLoader(name(), simpleName(), metrics),
+            new CompositeSyntheticFieldLoader.MalformedValuesLayer(name())
+        );
```
Comment (kkrik-es) [inline x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:723]: Cute :)

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java [hunk 4]
```diff
@@ -779,7 +779,7 @@ public void write(XContentBuilder b) throws IOException {
             if (metricHasValue.isEmpty()) {
                 return;
             }
-            b.startObject(simpleName);
```
Comment (kkrik-es) [inline x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:782]: Similar question, why do we need this?

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/aggregate-metrics/100_synthetic_source.yml [hunk 5]
```diff
@@ -102,3 +112,12 @@ aggregate_metric_double with ignore_malformed:
...
           value_count: 50
 
+  - do:
+      get:
+        index: test
+        id: "2"
+
+  - match:
+      _source:
+        metric: [{"min": 18.2,"max": 100.0, "value_count": 1}, "hey", 123, 456]
```
Comment (kkrik-es) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/aggregate-metrics/100_synthetic_source.yml:122]: Similar comment about missing the `[123, 456]` pair.

**Review by kkrik-es — APPROVED** [review]
👍

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java (hunk 2 above)
Comment (lkts) [reply x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:689]: I'll leave it if you don't mind i don't think it's too bad of a repetition.

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java (hunk 4 above)
Comment (lkts) [reply x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:782]: See #109882.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/aggregate-metrics/100_synthetic_source.yml (hunk 5 above)
Comment (lkts) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/aggregate-metrics/100_synthetic_source.yml:122]: See #109882.

**Pull request merged**